File: app/src/main/java/com/example/networkaplication/details/DetailsContract.java
package com.example.networkaplication.details;

import android.graphics.Bitmap;
import android.net.Uri;
import android.widget.ImageView;
import com.example.networkaplication.persistance.model.Details;

public interface DetailsContract {

    interface DetailsView {

        Bitmap getImage();

        void setImage(Uri imageUri);

        void initDetails(Details details);

        ImageView getImageView();
      
        void saveIsOffline();

        void backClicked();

        void showWebView();

        void showBrowser();

        void showAlertDialog();
    }

    interface DetailsPresenter {
      
        boolean onOptionItemSelected(int itemId);

        void initView(String filmName);

        void initViewById(String omdbId);

        void onSaveImageClicked(String omdbId);

        boolean isOffline();
    }


    interface DetailsRepository {

        void saveImage(Bitmap imageToSave, String omdbId);

        Uri loadImage(String omdbId);

        void loadDetails(String filmName);

        Details getDetailFromId(String omdbId);
    }
}

File: app/src/main/java/com/example/networkaplication/details/DetailsContract.kt
package com.example.networkaplication.details

import android.graphics.Bitmap
import android.net.Uri
import android.widget.ImageView

import com.example.networkaplication.persistance.model.Details

interface DetailsContract {

    interface DetailsView {

        val image: Bitmap

        val imageView: ImageView

        fun setImage(imageUri: Uri)

        fun initDetails(details: Details)

        fun saveIsOffline()

        fun backClicked()

        fun showWebView()

        fun showBrowser()

        fun showAlertDialog()

        fun getPermission()
    }

    interface DetailsViewModel {

        var isOffline: Boolean

        fun onOptionItemSelected(itemId: Int): Boolean

        fun initView(filmName: String)

        fun initViewById(omdbId: String)

        fun onSaveImageClicked(omdbId: String)
    }


    interface DetailsRepository {

        fun saveImage(imageToSave: Bitmap, omdbId: String)

        fun loadImage(omdbId: String): Uri?

        fun loadDetails(filmName: String)

        fun getDetailFromId(omdbId: String): Details
    }
}
